pkg/plugin/keyvault.go

```diff
@@ -102,29 +123,45 @@ func newKeyVaultClient(
 		keyVersion:       keyVersion,
 		vaultURL:         *vaultURL,
 		azureEnvironment: env,
+		keyID:            fmt.Sprintf("%x", sha256.Sum256([]byte(fmt.Sprintf("%s/keys/%s/%s", *vaultURL, keyName, keyVersion)))),
+		algorithm:        kv.RSA15,
 	}
 	return client, nil
 }
 
-func (kvc *keyVaultClient) Encrypt(ctx context.Context, cipher []byte) ([]byte, error) {
+func (kvc *keyVaultClient) Encrypt(ctx context.Context, cipher []byte) (*service.EncryptResponse, error) {
 	value := base64.RawURLEncoding.EncodeToString(cipher)
 
 	params := kv.KeyOperationsParameters{
-		Algorithm: kv.RSA15,
+		Algorithm: kvc.algorithm,
 		Value:     &value,
 	}
 	result, err := kvc.baseClient.Encrypt(ctx, kvc.vaultURL, kvc.keyName, kvc.keyVersion, params)
 	if err != nil {
 		return nil, fmt.Errorf("failed to encrypt, error: %+v", err)
 	}
-	return []byte(*result.Result), nil
+
+	annotations := map[string][]byte{
+		dateAnnotationKey:           []byte(result.Header.Get(dateAnnotationValue)),
+		requestIDAnnotationKey:      []byte(result.Header.Get(requestIDAnnotationValue)),
+		keyvaultRegionAnnotationKey: []byte(result.Header.Get(keyvaultRegionAnnotationValue)),
+		versionAnnotationKey:        []byte(encryptionResponseVersion),
+		algorithmAnnotationKey:      []byte(kvc.algorithm),
+		keyIDAnnotationKey:          []byte(kvc.keyID),
```

Comment:
We do not need the key ID twice ?

```suggestion
```

Reply:
We are using it in Decrypt to validate annotations.